st2actions/st2actions/worker.py

```diff
@@ -134,7 +138,21 @@ def process(self, liveaction):
 
     def shutdown(self):
         super(ActionExecutionDispatcher, self).shutdown()
-        # Abandon running executions if incomplete
+
+        # Check if there are other runners in service registry
+        coordinator = coordination.get_coordinator()
+        member_ids = []
+        service = "actionrunner"
+
+        try:
+            member_ids = list(coordinator.get_members(service.encode("utf-8")).get())
+        except GroupNotCreated:
+            pass
+
+        while member_ids and self._running_liveactions:
```

Comment:
Per my question / comment above - it appears we may need to utilize locking and "refresh" ``member_ids`` on each iteration - since that list could change during the time we sleep and wait.

Reply:
Moved refresh inside the loop